### ISSUE TITLE
False positive error with oas3-schema rule

### ISSUE THREAD

**Issue by MarcusAckermann** [stoplightio/spectral#1715]
**To Reproduce**

1. Given the attached OpenAPI/AsyncAPI document
2. Run this CLI command 'spectral lint openapi.json'
3. I get ' 38:18    error  oas3-schema       Property `schema` is not expected to be here.           components.headers.Cache-Control.schema'

**Expected behavior**
According to https://swagger.io/specification/#header-object the property 'schema' is allowed in a header declaration. There should be no error regarding rule oas3-schema.

[openapi.zip](https://github.com/stoplightio/spectral/files/6770087/openapi.zip)

**Comment by P0lip** [comment]
Hey, first of all, my apologies for a delayed response.
In our case, the way you reference that header object is invalid.
One should do

```json
     "headers": {
         "Cache-Control"  : {
                "$ref": "#/components/headers/Cache-Control"
          }
    }
```
I do agree however that the error could be more meaningful.

**Comment by jstockdi-marstone** [comment]
I came across this error and it stumped me for a bit.  For anyone else who has the same issue:

Change:
```
            "headers": {
              "Cache-Control" :{
                "schema": {
                  "$ref": "#/components/headers/Cache-Control"
                }
              }
            }
```

To:
```
            "headers": {
              "Cache-Control" :{
                "$ref": "#/components/headers/Cache-Control"
              }
            }
```

It's easy to mix up, as schema often has an attribute $ref in other use cases.  If the component has schema in it, then the object referencing it should not refer to it from a schema attribute.  I think... if anyone posts additional info here, that'd help me out.

NOTE:  only happens for version 3.0.1, it works fine in 3.1.0.
